integtest/src/main/java/org/corfudb/universe/cluster/docker/FakeDns.java

```diff
@@ -0,0 +1,178 @@
+package org.corfudb.universe.cluster.docker;
+
+import com.google.common.base.Throwables;
+import com.google.common.net.InetAddresses;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Fake DNS resolver which allows our tests to work well even though we use
+ * strange loopback IP addresses (127.x.y.z) with no corresponding reverse
+ * DNS.
+ * <p>
+ * This overrides the reverse lookups for such IPs to return the same address
+ * in String form.
+ * <p>
+ * Without this class, reverse DNS lookups for such addresses often take
+ * 5 seconds to return, causing timeouts and overall test slowness.
+ * <p>
+ * In the future this class might also be extended to test more interesting
+ * DNS-related scenarios.
+ */
+public class FakeDns {
+    private static FakeDns instance = new FakeDns();
+
+    private final Map<String, InetAddress> forwardResolutions = new HashMap<>();
+
+    private final Map<InetAddress, String> reverseResolutions = new HashMap<>();
+
+    /**
+     * whether the fake resolver has been installed
+     */
+    private boolean installed = false;
+
+    private FakeDns() {
+    }
+
+    public static FakeDns getInstance() {
+        return instance;
+    }
+
+    public synchronized void addForwardResolution(String hostname, InetAddress ip) {
+        forwardResolutions.put(hostname, ip);
+    }
+
+    public synchronized void addReverseResolution(InetAddress ip, String hostname) {
+        reverseResolutions.put(ip, hostname);
+    }
+
+    /**
+     * Install the fake DNS resolver into the Java runtime.
+     */
+    public synchronized void install() {
+        if (installed) return;
+        try {
+            try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)